pkg/ccl/backupccl/backup_planning.go

```diff
@@ -828,6 +836,28 @@ func backupPlanHook(
 	return fn, jobs.BulkJobExecutionResultHeader, nil, false, nil
 }
 
+func logAndSanitizeKmsURIs(ctx context.Context, kmsURIs ...string) error {
+	for _, dest := range kmsURIs {
+		clean, err := cloud.RedactKMSURI(dest)
+		if err != nil {
+			return err
+		}
+		log.Ops.Infof(ctx, "backup planning to connect to KMS destination %v", redact.Safe(clean))
+	}
+	return nil
+}
+
+func logAndSanitizeBackupDestinations(ctx context.Context, backupDestinations ...string) error {
+	for _, dest := range backupDestinations {
+		clean, err := cloud.SanitizeExternalStorageURI(dest, nil)
+		if err != nil {
+			return err
```

Comment:
I am comfortable making that change in both backports, this whole strategy is "best effort" at the moment anyways, as you pointed out in the other backport:
>I still have a bit of a misgiving about logging being an API of sorts here. It will be rather easy for us to break.

But im curious, why is failing hard okay in the master branch PR, but not in the backports? Is there some known issues w/ KMS pre 23.1 ?